src/connection.rs

```diff
@@ -221,6 +221,10 @@ pub fn connect(peer_contact: StaticContactInfo,
                     match deserialise::<ListenerResponse>(&read_buf[..bytes_rxd]) {
                         Ok(ListenerResponse::Connect { our_info, their_info, pub_key, }) => {
                             let (their_info, our_info) = (our_info, their_info);
+                            let their_pub_key = pub_key;
```

Comment:
Sorry, you are right. For some reason I though that it's a tuple, not struct. My bad.